src/span.cpp

```diff
@@ -148,8 +148,14 @@ void Span::FinishWithOptions(
   }
   tag = span_->meta.find(datadog_error_tag);
   if (tag != span_->meta.end()) {
-    span_->error = std::stoi(tag->second);
-    span_->meta.erase(tag);
+    // tag->second is the JSON-serialized value of the variadic type given to SetTag. If it's
+    // clearly falsey then set the error flag.
+    if (tag->second == "0" || tag->second == "" || tag->second == "false") {
+      span_->error = 0;
+    } else {
```

Comment:
Would it be safer at all to do it the other way around, look for truey values and fall back to false if it didn't match?

Reply:
discussed above